### PR TITLE
mpvScripts.mpv-subtitle-lines: init at 0-unstable-2024-05-19

### PR DESCRIPTION
**Pull request by purrpurrn**
## Description of changes
Added the mpv script mpv-subtitle-lines

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/applications/video/mpv/scripts/mpv-subtitle-lines.nix [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+  src = fetchFromGitHub {
+    owner = "christoph-heinrich";
+    repo = "mpv-subtitle-lines";
+    rev = "f0a85c7ba3370b490c46a77dcb2e212c0b149a50";
+    hash = "sha256-m9LHIz/hAOEq0DzHY3Jd1LM1cfxZng6Iaqrug1PfnAE=";
+  };
+
+  passthru.updateScript = unstableGitUpdater { };
+
+  meta = with lib; {
```
Comment (donovanglover) [inline pkgs/applications/video/mpv/scripts/mpv-subtitle-lines.nix:21]:
```suggestion
  meta = {
```
Avoid `with lib;`

File: pkgs/applications/video/mpv/scripts/mpv-subtitle-lines.nix [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    description = "List and search subtitle lines of the selected subtitle track";
+    homepage = "https://github.com/christoph-heinrich/mpv-subtitle-lines";
+    license = licenses.gpl2Only;
```
Comment (donovanglover) [inline pkgs/applications/video/mpv/scripts/mpv-subtitle-lines.nix:24]:
```suggestion
    license = lib.licenses.gpl2Only;
```

**Review by donovanglover — COMMENTED** [review]
LGTM after minor changes

**Review by donovanglover — APPROVED** [review]
No worries LGTM

**Comment by donovanglover** [comment]
Do we need to add the dependency [uosc](https://github.com/tomasklaen/uosc)? Seems like none of the existing scripts depend on other scripts.

**Comment by uninsane** [comment]
tested with
```
$ nix-shell -p 'mpv.override { scripts = [ mpvScripts.uosc mpvScripts.mpv-subtitle-lines ]; }'
$ mpv /path/to/media
```
then press ``` ` ``` and type `script-binding subtitle_lines/list_subtitles` 👍

**Pull request merged**